firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewModel.swift

```diff
@@ -0,0 +1,144 @@
+
+    // MARK: Accessibility Identifiers
+    let foxImageA11yId: String = AccessibilityIdentifiers.EnhancedTrackingProtection.MainScreen.foxImage
+    let shieldImageA11yId: String = AccessibilityIdentifiers.EnhancedTrackingProtection.MainScreen.shieldImage
+    let lockImageA11yId: String = AccessibilityIdentifiers.EnhancedTrackingProtection.MainScreen.lockImage
+    let arrowImageA11yId: String = AccessibilityIdentifiers.EnhancedTrackingProtection.MainScreen.arrowImage
+
+    let settingsA11yId = AccessibilityIdentifiers.EnhancedTrackingProtection.MainScreen.trackingProtectionSettingsButton
+    let domainLabelA11yId = AccessibilityIdentifiers.EnhancedTrackingProtection.MainScreen.domainLabel
+    let domainHeaderLabelA11yId = AccessibilityIdentifiers.EnhancedTrackingProtection.MainScreen.domainHeaderLabel
+    let statusTitleLabelA11yId = AccessibilityIdentifiers.EnhancedTrackingProtection.MainScreen.statusTitleLabel
+    let statusBodyLabelA11yId = AccessibilityIdentifiers.EnhancedTrackingProtection.MainScreen.statusBodyLabel
+    let trackersBlockedLabelA11yId = AccessibilityIdentifiers.EnhancedTrackingProtection.MainScreen.trackersBlockedLabel
+    let securityStatusLabelA11yId = AccessibilityIdentifiers.EnhancedTrackingProtection.MainScreen.securityStatusLabel
+    let toggleViewTitleLabelA11yId = AccessibilityIdentifiers.EnhancedTrackingProtection.MainScreen.toggleViewTitleLabel
+    let toggleViewBodyLabelA11yId = AccessibilityIdentifiers.EnhancedTrackingProtection.MainScreen.toggleViewBodyLabel
+    let closeButtonA11yId = AccessibilityIdentifiers.EnhancedTrackingProtection.MainScreen.closeButton
+    let faviconImageA11yId = AccessibilityIdentifiers.EnhancedTrackingProtection.MainScreen.faviconImage
+
+    var websiteTitle: String {
+        return url.baseDomain ?? ""
+    }
+
+    var connectionStatusString: String {
+        return connectionSecure ?
+            .Menu.EnhancedTrackingProtection.connectionSecureLabel :
+            .Menu.EnhancedTrackingProtection.connectionUnsecureLabel
+    }
+
+    var  connectionDetailsTitle: String {
+        let titleOn = String(format: String.Menu.EnhancedTrackingProtection.onTitle, AppName.shortName.rawValue)
+        return connectionSecure ? titleOn : .Menu.EnhancedTrackingProtection.offTitle
+    }
+
+    var  connectionDetailsHeader: String {
+        return connectionSecure ?
+            .Menu.EnhancedTrackingProtection.onHeader : .Menu.EnhancedTrackingProtection.offHeader
+    }
+
+    var  connectionDetailsImage: UIImage? {
+        return connectionSecure ?
+        UIImage(named: ImageIdentifiers.TrackingProtection.protectionOn) :
+        UIImage(named: ImageIdentifiers.TrackingProtection.protectionOff)
+    }
+
+    var isSiteETPEnabled: Bool {
+        switch contentBlockerStatus {
+        case .noBlockedURLs, .blocking, .disabled: return true
+        case .safelisted: return false
+        }
```

Comment:
could we also support this work with some tests? This might be a good simple area to check.

Reply:
I wanted to add the tests in another PR to keep them under 300 lines of code

Reply:
okay sounds good, I agree to keep PR small. In fact, I'll prob prefer this logic added in another PR so it's easier to read the changes and tests together, but not a blocker!